llm/server.go

```diff
@@ -215,6 +215,10 @@ func NewLlamaServer(gpus gpu.GpuInfoList, model string, ggml *GGML, adapters, pr
 
 	if opts.NumThread > 0 {
 		params = append(params, "--threads", strconv.Itoa(opts.NumThread))
+	} else if runtime.GOOS == "linux" {
+		if cpus, err := gpu.ParseCpuInfo(); err == nil {
```

Comment:
Would `runtime.NumCPU()` yield the same results here?

Reply:
Go reports the number of logical processors, without exposing if there is hyper-threading involved.  It also tops out at 64 on windows, even if there are more cores/threads on the system.